server/consumer.go

```diff
@@ -330,7 +330,25 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 
 	// Make sure any partition subject is also a literal.
 	if config.FilterSubject != _EMPTY_ {
-		if !mset.validSubject(config.FilterSubject) {
+		streams, cfg, valid := []*stream{mset}, mset.config(), false
```

Comment:
we cant lookup or verify this for streams with external set right? So I think we should handle those here

Reply:
I also need to do clustered version which I will work on now.